### ISSUE TITLE
Automated Block Rewards | zkApp

### ISSUE THREAD

**Issue by halsaphi** [MinaProtocol/mina#13299]
As a Block Producer and SPO I would like to automate my reward payouts to avoid time consuming manual work.

Build out the functionality and tooling that will enable staking rewards to be distributed automatically to those Mina users who delegate to node operators, or Block Producers.  This will result in less operational overhead for the Block Producer community, and more assurance and timeliness for delegators on their rewards payouts.

**Comment by deepthiskumar** [comment]
Two different routes that could be explored
1. Using a zkApp from which payouts can be retrieved (requires https://github.com/MinaProtocol/mina/issues/11436)
2. Possibly baked into consensus layer and then using zkApps to retrieve payout

**Comment by dannywillems** [comment]
Linking links we have been sharing around:
- [YouTube - Video - ZkAppPool](https://www.youtube.com/watch?v=maoA_5MgOA8&list=PLKIvwYrcKk8DdDgx5jBpEDoRyKWVP6EKM&index=13&t=1s) by Gareth Davis
- [HackMD - MINA zkApp staking pool](https://hackmd.io/@garethtdavies/BJH3xMpFs)
- Script to automate payout: https://github.com/jrwashburn/mina-pool-payout
- [zkAppPool - A zkApp based staking pool for MINA](https://github.com/garethtdavies/zkAppPool)
- An internal discussion on [o1Labs Slack](https://o1-labs.slack.com/archives/C01SJSXSM7H/p1752749557982739).

**Comment by dannywillems** [comment]
A suggestion from @bkase is to use Cosmos SDK's design, see https://docs.cosmos.network/v0.53/build/modules/distribution. I like the approach.

TL;DR: delegators make request to withdraw (on-demand delegator process). Staking rewards are kept in the pool until requested.

**Comment by dannywillems** [comment]
I know that AuroWallet uses a zkApp to pay staking rewards. Does anyone have a link to their work?
